integration-tests/test/native-eth-ovm-calls.spec.ts

```diff
@@ -0,0 +1,309 @@
+
+      expect(success).to.be.true
+      expect(BigNumber.from(returndata)).to.deep.eq(BigNumber.from(sendAmount))
+    })
+
+    it('should have the correct callvalue but not persist the transfer if the target reverts', async () => {
+      const sendAmount = 15
+      const internalCalldata = ValueCalls1.interface.encodeFunctionData(
+        'verifyCallValueAndRevert',
+        [sendAmount]
+      )
+      const [success, returndata] = await ValueCalls0.callStatic.sendWithData(
+        ValueCalls1.address,
+        sendAmount,
+        internalCalldata
+      )
+
+      expect(success).to.be.false
+      expect(returndata).to.eq(encodeSolidityRevertMessage('expected revert'))
+
+      await checkBalances([initialBalance0, 0])
+    })
+
+    it('should look like the subcall reverts with no data if value exceeds balance', async () => {
+      const sendAmount = initialBalance0 + 1
+      const internalCalldata = ValueCalls1.interface.encodeFunctionData(
+        'verifyCallValueAndReturn',
+        [sendAmount] // this would be correct and return successfuly, IF it could get here
+      )
+      const [success, returndata] = await ValueCalls0.callStatic.sendWithData(
+        ValueCalls1.address,
+        sendAmount,
+        internalCalldata
+      )
+
+      expect(success).to.be.false
+      expect(returndata).to.eq('0x')
+    })
+
+    it('should preserve msg.value through ovmDELEGATECALLs', async () => {
+      const Factory__ValueContext = await ethers.getContractFactory(
+        'ValueContext',
+        wallet
+      )
+      const ValueContext = await Factory__ValueContext.deploy()
+      await ValueContext.deployTransaction.wait()
+
+      const sendAmount = 10
+
+      const [
+        outerSuccess,
+        outerReturndata,
+      ] = await ValueCalls0.callStatic.sendWithData(
+        ValueCalls1.address,
+        sendAmount,
+        ValueCalls1.interface.encodeFunctionData('delegateCallToCallValue', [
+          ValueContext.address,
+        ])
+      )
+      const [
+        innerSuccess,
+        innerReturndata,
+      ] = ValueCalls1.interface.decodeFunctionResult(
+        'delegateCallToCallValue',
+        outerReturndata
+      )
+      const delegatedOvmCALLVALUE = ValueContext.interface.decodeFunctionResult(
+        'getCallValue',
+        innerReturndata
+      )[0]
+
+      expect(outerSuccess).to.be.true
+      expect(innerSuccess).to.be.true
+      expect(delegatedOvmCALLVALUE).to.deep.eq(BigNumber.from(sendAmount))
+    })
+
+    describe('Intrinsic gas for ovmCALL types', async () => {
+      let CALL_WITH_VALUE_INTRINSIC_GAS
+      let ValueGasMeasurer: Contract
+      before(async () => {
+        // Grab public variable from the EM
+        const OVM_ExecutionManager = new Contract(
+          await env.addressManager.getAddress('OVM_ExecutionManager'),
+          getContractInterface('OVM_ExecutionManager', false),
+          env.l1Wallet.provider
+        )
+        CALL_WITH_VALUE_INTRINSIC_GAS = (
```

Comment:
Style nit: nested awaits are bad, this could be 2 lines instead of 3 without a nested await